### PR TITLE
Removed some hardcoded move IDs

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Harcoded IDs removed:
- `MOVE_BLEAKWIND_STORM`
- `MOVE_WILDBOLT_STORM`
- `MOVE_SANDSEAR_STORM`
- `MOVE_BLIZZARD`
- `MOVE_FRUSTRATION`
- `MOVE_SELF_DESTRUCT`
- `MOVE_EXPLOSION`

For this, the following changes have been made:
- Added `rainAlwaysHit`, `hailAlwaysHit` and `snowAlwaysHit` move flags to always hit in rain, hail and snow respectively.
- Added `sunAccuracy50` move flag to set accuracy to 50 in sun.
- Removed `EFFECT_HURRICANE` and `EFFECT_THUNDER`. They now use `EFFECT_CONFUSE_HIT` and `EFFECT_PARALYZE_HIT` instead.
  - Modified `IsDomeComboMoveEffect` to account for these changes

Also
- Moved `argument` to reduce padding.

## Feature(s) this PR does NOT handle:
There's still a lot of hardcoded moves that this one still doesn't handle. Future PR potential.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: include/pokemon.h [hunk 1]
```diff
@@ -507,8 +508,10 @@ struct BattleMove
     u32 parentalBondBanned:1;
     u32 skyBattleBanned:1;
     u32 sketchBanned:1;
-
-    u16 argument;
+    u32 rainAlwaysHit:1;
+    u32 hailAlwaysHit:1;
+    u32 snowAlwaysHit:1;
+    u32 sunAccuracy50:1;
```
Comment (LOuroboros) [inline include/pokemon.h:514]:
```suggestion
    u32 sunAccuracyHalved:1;
```
Slightly more verbose but reads more nicely and is consistent with the labels of the other flags.

Comment (AsparagusEduardo) [reply include/pokemon.h:514]: Thought about it, but that wording implies that 70 accuracy would go to 35 accuracy.

Comment (LOuroboros) [reply include/pokemon.h:514]: Fair enough, it *is* hardcoded to 50 too and something like "sunAccuracyLowered" isn't descriptive enough considering that. Yuck. I hate Game Freak sometimes.

**Comment by AlexOn1ine** [comment]
Could we wait with merging this PR #3577 is merge?

**Comment by AsparagusEduardo** [comment]
> Could we wait with merging this PR #3577 is merge?

Sure thing

**Comment by AlexOn1ine** [comment]
> Could we wait with merging this PR after #3577 is merged?

So my suggestion would be to keep using `EFFECT_THUNDER` for both Thunder and Hurricane, make an `EFFECT_BLIZZARD` for blizzard and `EFFECT_FORCES_OF_NATURE` effect for the winds moves. 

I feel like using flags for this might be overkill since we have the effect field free now.

**Comment by AsparagusEduardo** [comment]
> So my suggestion would be to keep using `EFFECT_THUNDER` for both Thunder and Hurricane, make an `EFFECT_BLIZZARD` for blizzard

I suppose.
> and `EFFECT_FORCES_OF_NATURE` effect for the winds moves.

Not a fan of this proposed name. Enamorus' move is excluded and it doesn't feel right to call the effect the same as a group of mon.

**Comment by AlexOn1ine** [comment]
> > So my suggestion would be to keep using `EFFECT_THUNDER` for both Thunder and Hurricane, make an `EFFECT_BLIZZARD` for blizzard
> 
> I suppose.
> 
> > and `EFFECT_FORCES_OF_NATURE` effect for the winds moves.
> 
> Not a fan of this proposed name. Enamorus' move is excluded and it doesn't feel right to call the effect the same as a group of mon.

I'm fine with either name you choose.

**Comment by AsparagusEduardo** [comment]
I think I want to redo this PR (would force-push, but I don't think that's good ^^;)

**Pull request closed without merging**